src/shell/Session.ts

```diff
@@ -22,7 +22,7 @@ export class Session extends EmitterWithUniqueID {
     history = History;
     historicalPresentDirectoriesStack = new OrderedSet<string>();
 
-    constructor(private application: ApplicationComponent, private _dimensions: Dimensions) {
+    constructor(private application: ApplicationComponent, private _dimensions: Dimensions, public position: Positions) {
```

Comment:
I was thinking of storing sessions not as an array, but as a tree. And then during a tree traversal we'd know it's position.